File: 1518.py
class Solution:
    def numWaterBottles(self, numBottles: int, numExchange: int) -> int:
        res = 0
        buff = 0
        while numBottles > 0:
            res += numBottles
            numBottles, buff = divmod(numBottles + buff, numExchange)
        return res

File: sol/1518/1518.cpp
struct Solution {
    inline int numWaterBottles(int numBottles, int numExchange) {
        int res = 0, buff = 0;
        while (numBottles > 0){
            res += numBottles;
            numBottles += buff;
            buff = numBottles % numExchange;
            numBottles = numBottles / numExchange;
        }
        return res;
    }
};
